projects/common/vk_renderer.cpp

```diff
@@ -3453,6 +3457,283 @@ HRESULT CompileHLSL(
+    {
+        return COMPILE_ERROR_INTERNAL_COMPILER_ERROR;
+    }
+
+    slang::TargetDesc targetDesc = {};
+    targetDesc.format            = SLANG_SPIRV;
+    targetDesc.flags = SLANG_TARGET_FLAG_GENERATE_SPIRV_DIRECTLY;
+    //
+    // Disable profile for now since it's causing ISession::loadModuleFromSourceString to crash.
+    //
+    //targetDesc.profile           = globalSession->findProfile(profile.c_str());
+
+    targetDesc.flags |= SLANG_TARGET_FLAG_GENERATE_WHOLE_PROGRAM;
+
+    // Must be set in target desc for now
+    targetDesc.forceGLSLScalarBufferLayout = true;
+
+    // Compiler options for Slang
+    std::vector<slang::CompilerOptionEntry> compilerOptions;
+    {
+        // Define __spirv__
+        {
+            slang::CompilerOptionEntry entry = {slang::CompilerOptionName::MacroDefine};
+            entry.value.kind                 = slang::CompilerOptionValueKind::String;
+            entry.value.stringValue0         = "__spirv__";
+            entry.value.stringValue1         = "1";
+
+            compilerOptions.push_back(entry);
+        }
+
+        // Force Slang language to prevent any accidental interpretations as GLSL or HLSL
+        {
+            slang::CompilerOptionEntry entry = {slang::CompilerOptionName::Language};
+            entry.value.stringValue0         = "slang";
+
+            compilerOptions.push_back(entry);
+        }
+
+        // Force "main" entry point if requested
+        if (!options.ForceEntryPointMain)
+        {
+            compilerOptions.push_back(
+                slang::CompilerOptionEntry{
+                    slang::CompilerOptionName::VulkanUseEntryPointName,
+                    slang::CompilerOptionValue{slang::CompilerOptionValueKind::Int, 1}
+            });
+        }
+
+        // Use unscoped enums
+        {
+            compilerOptions.push_back(
+                slang::CompilerOptionEntry{
+                    slang::CompilerOptionName::UnscopedEnum,
+                    slang::CompilerOptionValue{slang::CompilerOptionValueKind::Int, 1}
+            });
+        }
+
+        // Force scalar block layout - this gets overwritten by forceGLSLScalarBufferLayout in
+        // the target desc currently. So we just set it there.
+        //
+        {
+            compilerOptions.push_back(
+                slang::CompilerOptionEntry{
+                    slang::CompilerOptionName::GLSLForceScalarLayout,
+                    slang::CompilerOptionValue{slang::CompilerOptionValueKind::Int, 1}
+            });
+        }
+    }
+
+    slang::SessionDesc sessionDesc       = {};
+    sessionDesc.targets                  = &targetDesc;
+    sessionDesc.targetCount              = 1;
+    sessionDesc.defaultMatrixLayoutMode  = SLANG_MATRIX_LAYOUT_COLUMN_MAJOR;
+    sessionDesc.compilerOptionEntries    = compilerOptions.data();
+    sessionDesc.compilerOptionEntryCount = static_cast<uint32_t>(compilerOptions.size());
```

Comment:
You could just use DataPtr(compilerOptions); and CountU32(compilerOptions); couldn't you?